### PR TITLE
Tensor naming convention review

### PR DESCRIPTION
**Pull request by dcslin**
1. added additional word to indicate mutation on underline data
2. add inplace flag to indicate mutation on underline data
3. updated cpp `sum()` on axis
4. no change to python `add_column()` or similar as it could be done by `broadcast` and could be deprecated.

### CODE REVIEW

File: python/singa/tensor.py [hunk 1]
```diff
@@ -232,14 +232,18 @@ def l1(self):
...
         Args:
             x (float), a float value to be set to all elements.
         '''
         # assert type(x) == float, 'set value only accepts float input'
         # if isinstance(x, float):
+        if not inplace:
+            # return new tensor filled with value
+            raise NotImplementedError
+
         self.data.SetFloatValue(float(x))
```
Comment (nudles) [inline python/singa/tensor.py]: return self if inplace is false; otherwise, return the new tensor?

File: python/singa/tensor.py [hunk 2]
```diff
@@ -352,74 +356,86 @@ def deepcopy(self):
...
             low (float): the lower bound
             high (float): the hight bound
         '''
+        if not inplace:
+            # return new tensor
+            raise NotImplementedError
+
         singa.Uniform(float(low), float(high), self.data)
 
     def add_column(self, v):
```
Comment (nudles) [inline python/singa/tensor.py]: use decorator  ```@deprecated ```

Comment (nudles) [reply python/singa/tensor.py]: print the the new function name in the msg.

Comment (dcslin) [reply python/singa/tensor.py]: thanks for the review, this is fixed.

File: python/singa/tensor.py (hunk 1 above)
Comment (dcslin) [reply python/singa/tensor.py]: thanks for the review, this is fixed

**Comment by dcslin** [comment]
Hi @nudles, thanks for the review, this is ready for merge.

File: python/singa/tensor.py (hunk 2 above)
Comment (dcslin) [reply python/singa/tensor.py]: reference https://github.com/apache/singa/pull/570/commits/203fcb91cd590d3436ce31a68844d8f47817be9c

File: python/singa/tensor.py (hunk 1 above)
Comment (dcslin) [reply python/singa/tensor.py]: reference https://github.com/apache/singa/pull/570/commits/8cc44683abfb21be284a751f501ba31eb4737022

**Pull request merged**